tests/nodejs/migrations.test.ts

```diff
@@ -12,7 +13,7 @@ import {
 for (const dialect of SUPPORTED_DIALECTS) {
   describe(`KyselyFsMigrationSource: ${dialect}`, () => {
     let ctx: TestContext
-    let migrationSource: KyselyFsMigrationSource
+    let migrationSource: FsMigrations
```

Comment:
Why was this change necessary?

Reply:
not necessary per-se, and independent of the core of this PR. it's a separate commit before the actual one that splits the bundle; figured if I'm already here, might as well clean up sth on my way out.

---

I noticed that TS was not happy with the `KyselyFsMigrationSource` type assignment here, complaining that:
```
let migrationSource: KyselyFsMigrationSource
                     ~~~~~~~~~~~~~~~~~~~~~~~
'KyselyFsMigrationSource' refers to a value, but is being used as a type here. Did you mean 'typeof KyselyFsMigrationSource'?
```
However, switching to `typeof KyselyFsMigrationSource` causes _even more TS problems_.

Using `InstanceType<typeof KyselyFsMigrationSource>` is _more accurate_ with the later assignment, but then TS still complains about this type not perfectly matching what `knex.migrate.rollback` expects.

Falling back to knex's own `FsMigrations` type aligns with the variable assignment, and with with knex itself later expects.

...just one of the cases were I do sympathize with those who hate TS and its type juggling. 😅

Reply:
I see. Let's go with `InstanceType<typeof KyselyFsMigrationSource>`.

Reply:
> I see. Let's go with InstanceType<typeof KyselyFsMigrationSource>.

roger, pushing this. as per my earlier post, this still causes TS complain in two places:
```
await ctx.knex.migrate.latest({migrationSource})
                               ~~~~~~~~~~~~~~~
Type 'KyselyFsMigrationSource' is missing the following properties from type 'MigrationSource<unknown>': getMigrations, getMigrationNamets(2739)
```
but these type errors should be safe to ignore.

> The build step turned the exported class declaration into var x = class.

can you clarify (if worth it)? the ESM output should not be relevant to the TS warning, and this TS type declaration here should have no impact on any build or test output.